conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/FeatureSwitches.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.interpreter
+
+/**
+ * Class bundling together switched for features to be used during Conformance
+ * It's a sealed abstract case class to enforce the the provided constructor (apply). This way values can be
+ * changed only via the setters and in case of removal of any of them, compiler will complain whenever they were used.
```

Comment:
The idea of the class to simplify adding similar switches without the need to rewrite the whole chain of functions using them. Or many of the tests, unless they use the new switch.
And by forcing the setting of the values via the setter only, it ensures that in future, when the particular switch is removed it will not cause miss-configuration. Imagine code:
`val fs = FeatureSwitches(false, true)` and now we don't need the `catalystWorkaroundEnabled` anymore so it's removed. The constructor approach would still pass the compiler but suddenly a wrong switch is set. The use of setters prevents this.
Will consider better doc. Suggestions are welcomed. 😃

Reply:
I see what you mean. If you declare them in the constructor as named params you get the same effect for free. But you can't enforce it so everyone will always remember to name the params, so it makes sense.